app/src/main/java/org/oppia/android/app/player/state/StateViewModel.kt

```diff
@@ -26,9 +49,28 @@ class StateViewModel @Inject constructor() : ObservableViewModel() {
   val isHintBulbVisible = ObservableField(false)
   val isHintOpenedAndUnRevealed = ObservableField(false)
 
+  val hasSupportForSwitchingToSwahili: Boolean = enableLearnerStudy.value
+  val hasSwahiliTranslations: LiveData<Boolean> by lazy {
```

Comment:
Just FYI (may not affect this code) -- our current Swahili translation support for Multiplication is around 40% due to changes in the lessons. Wanted to make sure that what we're implementing here accounts for the case where some strings have not been translated and that Swahili would still show up as an option despite 40% being much less than 100%.

Note that the Swahili language switcher is an optional requirement overall, if handling the above correctly (in terms of not causing breakages/crashes, or the user not being able to get back to the "base" state) is difficult in any way.

Reply:
The logic is currently implemented as "if there are *any* Swahili translations, show the switch button." We could easily change it to a percentage or something else--do you have any suggestions?

Reply:
I think the best would be to show the switch button on a card-by-card basis if the *content* part of that card has a Swahili translation. If the user subsequently navigates to a card whose content is not translated then the switch button should not be shown and the display should be in English. Likewise if the user navigates to a card whose content has a Swahili translation then the switch button should be shown.

I know there's a gap here where the content may be translated but e.g. the hints/solutions aren't, but I think that that is OK to live with because the major issues with comprehension tend to stem from students not understanding the question they are being asked.

On the other hand, if the switch button is at the lesson level and cannot easily be made into a per-card thing, then I would suggest removing it entirely, because otherwise you'll get students reading the card in English while the button says that the card is in Swahili, and this would be confusing.